spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -20,110 +20,126 @@ On Ethereum, there are two types of accounts: externally owned accounts, control
 
-`tryRegisterIBCAccount` method in `IBCAccountModule` interface defines the way to request the creation of an IBC account on the host chain (or counterparty chain that the IBC account lives on). The host chain creates an IBC account using its account creation logic, which may use data within the packet (such as destination port, destination channel, etc) and other additional data for the process. The origin chain can receive the address of the account that was created from the acknowledge packet.
+### Architecture Diagram
+![](https://i.imgur.com/HX1h2u2.png)
 
-`tryRunTx` method in `IBCAccountModule` interface defines the way to create an outgoing packet for a specific `type`. `Type` indicates how the IBC account transaction should be constructed and serialised for the host chain. Generally, `type` indicates what blockchain framework the host chain was built on.
 
-`createAccount` defines the way to determine the account's address by using the packet. If the host chain doesn't support a deterministic way to generate an address with data, it can be generated using the internal logic of the host chain. A newly created interchain account must not conflict with an existing account. Therefore, the host chain (on that the interchain account lives in) must keep track of which blockchains have created an interchain account within the host chain in order to verify the transaction signing authority in `authenticateTx`.
 
-`authenticateTx` validates the transaction and checks that the signers in the transaction have the right permissions. `
+### Authentication & Authorization
+The sending chain (the chain registering and controlling an account) will implement its own authentication and authorization, which will determine who can create an interchain account and what type of transactions the registered accounts can invoke. One example of this may be a cosmos SDK chain that only allows the creation of an interchain account on behalf of the chains distribution module (community pool), whereby actions this interchain account takes are determined by governance proposals voted on by the token holders of the sending chain. Another example may be a smart contract that registers an interchain account and has a specific set of actions it is authorized to take.  
 
-`runTx` executes a transaction after the transaction has been successfully authenticated.
+The receiving chain must implement authentication with regard to ensuring that the incoming messages are sent by the owner of the targeted interchain account. With regards to authorization, it is up to each chain-specific implementation to decide if the hosted interchain accounts have the authority to invoke all of the chain's message types or only a subset. This configuration may be set up at module initialization.
```

Comment:
I think @AdityaSripal might have good suggestions here, but I think it needs to be specified that the receiving chain authorization is dependent upon the sending chain authorization. The receiving chain should just be checking that the owners the sending chain authorized to send the messages are also authorized to execute those messages on the receiving chain. 

Ex:
Sending chain authorizes a packet which says addrA will send 100 atoms from addrA to addrB. Receiving chain should check that the owner (addrA) matches the from address (addrA) and that the spendable balance is correct. 

If the scenario was: Sending chain authorizes a packet which says addrA will send 100 atoms from addrC to addrB
Then the receiving chain should throw an error because addrA is not authorized to spend from addrC account. What cannot really happen is that the receiving chain verifies some signature by addrA over the messages being sent. This sort of verification should be done by the sending chain

Reply:
Thanks for the feedback. 

Just so we're on the same page: Authentication confirms that users are who they say they are. Authorization gives those users permission to access a resource.

Authentication & Authorization on the sending side determines who can create accounts, and what transactions those accounts can send. Every chain that wants to act as a sender will need to implement this logic. 

Authentication on the receiving side is simply checking is the sender of this message, the owner of the targetted account.

An additional, optional step, may be to add authorization on the receiving side,  by disallowing certain messages from being invoked by interchain accounts. This can just be an array of message types that get passed in at init time. Maybe, the creators of a chain do not want external chains to be able to access all message types via an interchain account. This was raised by B-harvest in a conversation.

Reply:
I see, thanks for the clarification. I think I was expecting to read the authentication on the receiving side before the additional, optional step as the authentication on the receiving side is non-optional

Reply:
Cool. I'll think about that and update it to be more clear. Thanks.

Reply:
I think it is clear and concise as is. It just took me a second pass/more detailed read. Looks good to me :+1: